apps/angular/projection/src/app/data-access/teacher.store.ts

```diff
@@ -1,23 +1,22 @@
-import { Injectable } from '@angular/core';
-import { BehaviorSubject } from 'rxjs';
+import { Injectable, computed, signal } from '@angular/core';
 import { Teacher } from '../model/teacher.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class TeacherStore {
-  private teachers = new BehaviorSubject<Teacher[]>([]);
-  teachers$ = this.teachers.asObservable();
+  private _teachers = signal<Teacher[]>([]);
+  teachers = computed(this._teachers);
```

Comment:
you can use _teachers.asReadonly() to create a readonly signal

Reply:
I dit it in my local code Thanks!